### DISCUSSION TITLE
Can I make type stubs for click-completions?

### DISCUSSION THREAD

**Discussion by mahyarmirrashed** [python/typing#1858] [Q&A] [answered]
The discussion forum for typeshed said to ask before embarking on significant work. I would contribute to the library itself, but the maintainer seems to be unresponsive.

https://github.com/click-contrib/click-completion

**Comment by hauntsaninja** [comment, marked as answer]
Sure, although if you're using click>=8 you should use the built-in completion https://click.palletsprojects.com/en/8.1.x/shell-completion/

**Reply by mahyarmirrashed** [reply]
Unfortunately, I have to use a lower version of Click. Otherwise, I would use the modern library!
Thanks though, I'm going to try to keep getting in contact with the completion library's maintainer. If they don't respond, I'll create the type stubs and PR them here.
